### ISSUE TITLE
Unsupported raw identifiers in 0.5

### ISSUE THREAD

**Issue by fujiapple852** [danburkert/prost#179]
I have a proto file (that I do not control) which contains a field named `self` (of type `Unit`).  In PROST 0.4 this is generated as `Self_` which works whereas in PROST 0.5 it is generated as `r#Self` which fails to compile in stable rust (1.34) with `r#Self is not currently supported` (see discussion on why [here](https://internals.rust-lang.org/t/raw-identifiers-dont-work-for-all-identifiers/9094/2)).

Workaround is either to use PROST 0.4 (which in turn forces using Rust 2015 and therefore the PROST generated code needs to be in a separate crate for Rust 2018 projects) or some clunky pre or post processing of the proto file or generated code.

Suggested fix: revert to appending an `_` to keyword identifiers for the list of 5 keywords not supported by raw identifiers.

Aside: I tried using the `field_attributes` option to prepend an `_` to the field but this doesn't work as it is rendered as a separate token to the identifier:

        #[prost(message, tag="1")]
        _
        r#Self(super::Unit),

**Comment by fujiapple852** [comment]
I made a simple local fix to the [to_snake](https://github.com/danburkert/prost/blob/master/prost-build/src/ident.rs#L7) and [to_upper_camel](https://github.com/danburkert/prost/blob/master/prost-build/src/ident.rs#L31) functions to resolve the issue.  

I'd be happy to submit a PR however i'm unclear on what the desired logic should be here given the current implementation.  Specifically for the `to_upper_camel` function which seems to deal only with the `Self` case explicitly at the moment:

```
/// Converts a `snake_case` identifier to an `UpperCamel` case Rust type identifier.
pub fn to_upper_camel(s: &str) -> String {
    let mut ident = s.to_camel_case();

    // Use a raw identifier if the identifier matches a Rust keyword:
    // https://doc.rust-lang.org/reference/keywords.html.
    if ident == "Self" {
        ident.insert_str(0, "r#");
    }
    ident
}
```

**Comment by danburkert** [comment]
Fixed in #180, thanks!
